src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -831,6 +879,28 @@ private static class ImageHelper implements ImageProvider {
                                 .collect(Collectors.toSet());
         }
 
+        // returns true if the current platform's "jmods" directory is the parent of the
+        // passed javaBasePath
+        private boolean isJavaBaseFromCurrentPlatform(Path javaBasePath) throws IOException {
+            Path currentPlatformJmods = getDefaultModulePath();
+            if (currentPlatformJmods == null) {
+                return false;
+            }
+            boolean ret;
+            try {
+                @SuppressWarnings("removal")
+                var unused = ret = AccessController.doPrivileged(
+                        // check if the current platform's "jmods" directory is the parent of
+                        // the "java.base" module file used to create the image
+                        (PrivilegedExceptionAction<Boolean>) () -> Files.isSameFile(javaBasePath,
+                                currentPlatformJmods.resolve(Path.of("java.base.jmod"))));
+            } catch (PrivilegedActionException e) {
```

Comment:
jlink doesn't run with a security manager so no need for the AccessController.doPriv.

Assuming the name of the jmod file is probably okay.

Reply:
> jlink doesn't run with a security manager so no need for the AccessController.doPriv.

Thank you for that detail - I've now removed the SecurityManager checks from this newly introduced code.

> Assuming the name of the jmod file is probably okay.

Were you thinking of exploded builds? I'm not too familiar with exploded builds, but after you commented here about the `.jmod` extension, I checked the existing code in `JLinkTask` and I can see that we seem to allow the module to be packaged as a `.jar` too, like here https://github.com/openjdk/jdk/blob/master/src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java#L834. Do you think this cross platform image creation would be relevant/used with exploded builds and if so, should I update the PR to address it?

Reply:
> Do you think this cross platform image creation would be relevant/used with exploded builds and if so, should I update the PR to address it?

Actually, now that I think about it, this `isJavaBaseFromCurrentPlatform()` method will get executed even for regular (same platform) image generation. Which means if the `java.base` module file doesn't end in a `.jmod` extension then this part of the code can run into `java.nio.file.NoSuchFileException` . I will generate an exploded build and see if anything needs to be done.

Reply:
I decided to remove the use of hardcoded `.jmod` extension while checking the paths. I have updated the PR to use a slightly different technique which doesn't require using hardcode file names for determining if this a cross-platform image generation.

Manual testing shows that this change continues to work fine when identifying cross-platform image generation. Existing automated tests too pass fine for same platform image generation.